fetchai/fetch.py

```diff
@@ -8,30 +20,55 @@ def ai(
         str
     ] = "proto:a03398ea81d7aaaf67e72940937676eae0d019f8e1d8b5efbadfef9fd2e98bb2",
 ) -> dict:
-    url = "https://agentverse.ai/v1/search/agents"
-    headers = {
-        "Content-Type": "application/json",
-    }
-
-    data = {
-        "search_text": query,
-        "sort": "relevancy",
-        "filters": {
-            "protocol_digest": [protocol],
-        },
-        # "geo_filter": {
-        #     "latitude": 52.19652,
-        #     "longitude": 0.1313,
-        #     "radius": 1000
-        # },
-        # Sort options: relevancy, created-at, last-modified, interactions
-        "direction": "asc",  # Ascending order; use "desc" for descending
-        "offset": 0,
-        "limit": 10,
-    }
+    agent_search_request = agentverse_client.search.AgentSearchRequest(
+        search_text=query,
+        filters=AgentFilters(protocol_digest=[protocol]),
+        sort=SortType.RELEVANCY,
+        direction=Direction.ASC,  # TODO: Should not be this DESC?
+        offset=0,
+        limit=10,
+    )
+
+    req: Callable[[SearchApi], AgentSearchResponse] = (
```

Comment:
why? It adds extra complication, function is less readable. Just use it direcly without this lambda and using extra method